x-pack/plugins/fleet/server/services/agent_policies/package_policies_to_agent_permissions.ts

```diff
@@ -101,8 +101,10 @@ export async function storedPackagePoliciesToAgentPermissions(
                       stream.compiled_stream?.data_stream?.dataset ?? stream.data_stream.dataset,
                   };
 
-                  if (stream.data_stream.elasticsearch) {
-                    ds.elasticsearch = stream.data_stream.elasticsearch;
+                  const pkgDefinedEs = dataStreams.find(d => d.type == ds.type && d.dataset == ds.dataset)?.elasticsearch;
+
+                  if (pkgDefinedEs) {
+                    ds.elasticsearch = pkgDefinedEs;
```

Comment:
I find it problematic that we were taking the package policy as source of truth here. I haven't tested it, but if someone can put arbitrary ES privileges on the API, the user could bypass the privilege requirements defined by the package. This change here prevents that.

Reply:
I decided to put this back and follow the existing pattern but I do think this is something we should cleanup. I've added a TODO in the code.